### PR TITLE
Update Kotlin to 1.9.21

### PR DESCRIPTION
**Pull request by eygraber**

### CODE REVIEW

File: build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts [hunk 1]
```diff
@@ -14,15 +13,22 @@ kotlin {
...
+    iosArm64()
+    iosX64()
+    iosSimulatorArm64()
+    tvosArm64()
+    tvosX64()
+    tvosSimulatorArm64()
+    watchosArm32()
+    watchosArm64()
+    watchosX64()
+    watchosSimulatorArm64()
```
Comment (eygraber) [inline build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts:28]: preset APIs were deprecated.

File: build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts [hunk 2]
```diff
@@ -14,15 +13,22 @@ kotlin {
...
 
     jvm()
 
-    @OptIn(ExperimentalKotlinGradlePluginApi::class)
-    targetHierarchy.default()
```
Comment (eygraber) [inline build-logic/src/main/kotlin/kotlin-inject.multiplatform.gradle.kts:24]: This is now the default and doesn't need to be explicitly called

File: kotlin-inject-runtime/src/nativeMain/kotlin/me/tatarka/inject/internal/LazyMap.kt [hunk 3]
```diff
@@ -1,10 +1,8 @@
 package me.tatarka.inject.internal
 
-import kotlin.native.concurrent.SharedImmutable
 import kotlinx.atomicfu.locks.ReentrantLock
 import kotlinx.atomicfu.locks.withLock
 
-@SharedImmutable
```
Comment (eygraber) [inline kotlin-inject-runtime/src/nativeMain/kotlin/me/tatarka/inject/internal/LazyMap.kt:7]: Deprecated, message says that it doesn't actually do anything

**Comment by evant** [comment]
> Also might be worth not merging until KSP has support for Kotlin 1.9.21 which should come concurrently with the fix for https://github.com/google/ksp/issues/1591

Yeah looks like the tests are failing because of this so think we'll have to wait

**Comment by evant** [comment]
https://github.com/google/ksp/releases/tag/1.9.21-1.0.15

**Comment by eygraber** [comment]
Updated

**Review by evant — APPROVED** [review]

**Pull request merged**